### PR TITLE
feat!: deprecate no-new-symbol, recommend no-new-native-nonconstructor

### LINKED ISSUE #17446: Change Request: deprecate `no-new-symbol`, recommend `no-new-native-nonconstructor`
### ESLint version

v8.46.0

### What problem do you want to solve?

The [`no-new-symbol`](https://eslint.org/docs/latest/rules/no-new-symbol) rule is not unique, because it reports a subset of the problems reported by the more recent [`no-new-native-nonconstructor`](no-new-native-nonconstructor) rule.

### What do you think is the correct solution?

This is a proposal to deprecate `no-new-symbol` and replace it with `no-new-native-nonconstructor` in the list of recommended rules.

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

This would be a breaking change per our [semantic version policy](https://github.com/eslint/eslint#semantic-versioning-policy), so it should be scheduled for the next major version.

### PR DESCRIPTION
**Pull request by fasttime**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[X] Other, please explain:

Fixes #17446 

#### What changes did you make? (Give an overview)

Deprecated and unrecommended rule `no-new-symbol`, added `no-new-native-nonconstructor` to recommended rules.

#### Is there anything you'd like reviewers to focus on?

See the comments. The respective issue #17446 is planned for the beta phase, but I'm not sure if the idea is to merge this in v9 or earlier.

### CODE REVIEW

File: docs/src/rules/no-new-symbol.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ further_reading:
 - https://www.ecma-international.org/ecma-262/6.0/#sec-symbol-objects
 ---
 
-
+This rule was **deprecated** in ESLint v8.54.0 and replaced by the [no-new-native-nonconstructor](no-new-native-nonconstructor) rule.
```
Comment (fasttime) [inline docs/src/rules/no-new-symbol.md:9]: This should be _deprecated in ESLint v9.0.0_, right?

File: lib/rules/no-new-symbol.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 /**
  * @fileoverview Rule to disallow use of the new operator with the `Symbol` object
  * @author Alberto Rodríguez
+ * @deprecated in ESLint v8.54.0
```
Comment (fasttime) [inline lib/rules/no-new-symbol.js:4]: Or _in ESLint v9.0.0_?

**Comment by mdjermanovic** [comment]
> The respective issue #17446 is planned for the beta phase, but I'm not sure if the idea is to merge this in v9 or earlier.

Yes, this is planned for v9 as we can't add new rules to `eslint:recommended` in a minor version. I'll mark this as breaking and convert it to draft just so that we don't accidentally merge it before we start with merging v9 PRs.

File: docs/src/rules/no-new-symbol.md (hunk 1 above)
Comment (mdjermanovic) [reply docs/src/rules/no-new-symbol.md:9]: Yes, v9.0.0.

File: lib/rules/no-new-symbol.js (hunk 2 above)
Comment (mdjermanovic) [reply lib/rules/no-new-symbol.js:4]: Yes, v9.0.0.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Review by nzakas — APPROVED** [review]
LGTM. Will merge for v9.0.0-alpha.

**Comment by mdjermanovic** [comment]
There are merge conflicts in `eslint-recommended.js`

**Pull request merged**